tests/discovery.js

```diff
@@ -257,3 +330,58 @@ test('discovery - mdns join, leave, join', async (t) => {
   t.ok(discover1.topics.length === 1, 'expected topic object')
   await discover2.join(keyPair.publicKey, { dht: false })
 })
+
+// Create a stress test with many peers to see if the connections made using the proper discovery mechanisms
+// i.e. mdns peers will only connect with mdns-enabled peers and dht peers will only connect with dht-enabled peers
+test('discovery - valid connection discovery types', async (t) => {
+  let connectionsCount = 0
+
+  const keyPair = createCoreKeyPair('stress test')
+
+  const instances = Array(20)
```

Comment:
out of curiousity I tried increasing this number and after about 75 the tests will start to hang without showing errors. not a blocker at all just kinda interesting!

we might want to explore whether that's a network limitation or a number of mdns/dht servers on a device limitation, but that's outside the scope of this pr. I think 20 is a good number for this.